src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValue.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotIgnoreReturnValueAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string CA2022RuleId = "CA2022";
+
+        internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRule = DiagnosticDescriptorHelper.Create(
+            CA2022RuleId,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessage)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.BuildWarning,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRuleWithMessage = DiagnosticDescriptorHelper.Create(
+            CA2022RuleId,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessageCustom)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.BuildWarning,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DoNotIgnoreReturnValueRule);
```

Comment:
Consider adding the other rule here. It wouldn't matter much since they have the same id, but I guess we always include all rules, even for duplicate IDs.